core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -308,7 +311,9 @@ public ServerTransportListener transportCreated(ServerTransport transport) {
       synchronized (lock) {
         transports.add(transport);
       }
-      return new ServerTransportListenerImpl(transport);
+      ServerTransportListenerImpl stli = new ServerTransportListenerImpl(transport);
+      stli.init();
```

Comment:
Is this the precise point that protocol negotiation starts, and transportReady the precise point that it ends? InProcessServer also does this? Does it make sense?

Reply:
At this point a TCP connection has been established, but nothing else. transportReady is called before any RPCs are triggered.